src/Session.php

```diff
@@ -168,13 +168,15 @@ public static function init(Auth $auth)
 
                     $auth->user->computePreferences();
                     foreach ($CFG_GLPI['user_pref_field'] as $field) {
-                        if ($field == 'language' && isset($_POST['language']) && $_POST['language'] != '') {
-                            $_SESSION["glpi$field"] = $_POST[$field];
-                        } else if (isset($auth->user->fields[$field])) {
+                        if (isset($auth->user->fields[$field])) {
                             $_SESSION["glpi$field"] = $auth->user->fields[$field];
                         }
                     }
 
+                    if (isset($_POST['language']) && $_POST['language'] != '' && isset($CFG_GLPI['languages'][$_POST['language']])) {
+                        $_SESSION["glpilanguage"] = $_POST['language'];
+                    }
```

Comment:
I guess the purpose of this is to be able to display a confirmation message in the target language when the user uses the language switch feature. This can probably be done in `User::prepareInputForUpdate()` when the target user is the current user.

It would prevent side effects, for instance when the `language` present in the `$_POST` variable is related to something else than the current user.